full-node/sov-prover-storage-manager/src/dummy_storage.rs

```diff
@@ -0,0 +1,157 @@
+use std::marker::PhantomData;
+
+use byteorder::{BigEndian, ReadBytesExt};
+use sov_schema_db::schema::{KeyDecoder, KeyEncoder, Result as CodecResult, ValueCodec};
+use sov_schema_db::snapshot::{DbSnapshot, QueryManager};
+use sov_schema_db::{define_schema, CodecError};
+use sov_state::MerkleProofSpec;
+
+/// Oversimplified representation of [`sov_state::ProverStorage`]
+pub struct NewProverStorage<Mps: MerkleProofSpec, Q> {
+    state_db: DbSnapshot<Q>,
+    native_db: DbSnapshot<Q>,
+    p: PhantomData<Mps>,
+}
+
+impl<Mps: MerkleProofSpec, Q: QueryManager> NewProverStorage<Mps, Q> {
+    pub(crate) fn with_db_handlers(
+        state_db_snapshot: DbSnapshot<Q>,
+        native_db_snapshot: DbSnapshot<Q>,
+    ) -> Self {
+        NewProverStorage {
+            state_db: state_db_snapshot,
+            native_db: native_db_snapshot,
+            p: Default::default(),
+        }
+    }
+
+    pub(crate) fn freeze(self) -> (DbSnapshot<Q>, DbSnapshot<Q>) {
+        let NewProverStorage {
+            state_db,
+            native_db,
+            ..
+        } = self;
+        (state_db, native_db)
+    }
+
+    #[cfg(test)]
+    pub(crate) fn read_state(&self, key: u64) -> anyhow::Result<Option<u64>> {
+        let key = DummyField(key);
```

Comment:
Would be nice not to mix `Dummy..` and prod code in one file. Would it be possible to make `state_db` `pub(crate)` and implemnthe these methods in test file?

Reply:
I will keep it this way for now, as it is going to be removed in the next PR.